shared_task_code/task_queue_consumer.py

```diff
@@ -32,32 +32,64 @@ def run_scan(self, scan, message_id):
         s3file = f"{message_id}-{date_string}-{self.task_name}.txt"
 
         results_filename = f"/tmp/{s3file}"
-
-        if self.func_taskcode != None:
-            self.func_taskcode(self.event, scan, message_id, results_filename, '/tmp/')
-
+        scan['start_time'] = f"{datetime.now():%Y-%m-%dT%H:%M:%S}Z"
+        if self.func_task_code != None:
+            self.func_task_code(self.event, scan, message_id, results_filename, '/tmp/')
+        scan['end_time'] = f"{datetime.now():%Y-%m-%dT%H:%M:%S}Z"
         subprocess.check_output(f'cd /tmp;tar -czvf "{s3file}.tar.gz" "{s3file}"', shell=True)
         s3 = boto3.resource("s3", region_name=self.region)
+
         s3.meta.client.upload_file(
             f"/tmp/{s3file}.tar.gz",
             self.event["ssm_params"][self.RESULTS],
             f"{s3file}.tar.gz",
             ExtraArgs={'ServerSideEncryption': "AES256", 'Metadata': scan})
 
-    def start(self, ValidateData=None, TaskCode=None):
-        # Pass in variables by name for:
-        # ValidateData=func() - optional (validates the event data)
-        # TaskCode=func() - the code to execute for this task
-        self.event['ssm_params'] = self.get_ssm_params(self.ssm_client, self.RESULTS)
+    def process_records(self, task_func, validate_data, queue_input):

```

Comment:
Why are we passing in a validate function? The more OO approach is to have our subclasses implement a validation function